### PR TITLE
[cmake][clad] Fix potential Clad build error

### PR DESCRIPTION
**Pull request by bellenot**
This should fix the following potential error when building clad:
```
  Command failed: 2
  ‘/usr/local/bin/cmake’ ‘–build’ ‘.’ ‘–config’ ‘.’
```

### CODE REVIEW

File: interpreter/cling/tools/plugins/clad/CMakeLists.txt [hunk 1]
```diff
@@ -19,6 +19,7 @@ if(MSVC)
   else()
     set(_clad_build_type Release)
   endif()
+  set(EXTRA_BUILD_ARGS --config ${_clad_build_type})
```
Comment (guitargeek) [inline interpreter/cling/tools/plugins/clad/CMakeLists.txt:22]: Are we sure that `_clad_build_type` is only defined in the case of Windows?
I see it's defined as `set(_clad_build_type ${CMAKE_CFG_INTDIR})`, and `CMAKE_CFG_INTDIR` seems to be defined also on XCode:
https://cmake.org/cmake/help/latest/variable/CMAKE_CFG_INTDIR.html

What about just adding a new if statement outside `if(MSVC)`:
```cmake
if(_clad_build_type)
   set(EXTRA_BUILD_ARGS --config ${_clad_build_type})
endif()
```

Comment (bellenot) [reply interpreter/cling/tools/plugins/clad/CMakeLists.txt:22]: You're right, I'll cross-check. Thanks Jonas!

Comment (Mouchemhou) [reply interpreter/cling/tools/plugins/clad/CMakeLists.txt]: hi bellenot,
I added , 
if(_clad_build_type STREQUAL "")
   set(EXTRA_BUILD_ARGS --config ${_clad_build_type})
endif()

is this the only modification should I make to this file?

Comment (bellenot) [reply interpreter/cling/tools/plugins/clad/CMakeLists.txt]: No, just take the diff:
```
diff --git a/interpreter/cling/tools/plugins/clad/CMakeLists.txt b/interpreter/cling/tools/plugins/clad/CMakeLists.txt
index c00f3ab3dc438..9bdc9bef0e521 100644
--- a/interpreter/cling/tools/plugins/clad/CMakeLists.txt
+++ b/interpreter/cling/tools/plugins/clad/CMakeLists.txt
@@ -20,6 +20,9 @@ if(MSVC)
     set(_clad_build_type Release)
   endif()
 endif(MSVC)
+if(NOT _clad_build_type STREQUAL "")
+  set(EXTRA_BUILD_ARGS --config ${_clad_build_type})
+endif()
 set(_CLAD_LIBRARY_PATH ${CMAKE_CURRENT_BINARY_DIR}/clad-prefix/src/clad-build/${_clad_build_type}/lib${LLVM_LIBDIR_SUFFIX})
 
 # build byproducts only needed by Ninja
@@ -86,8 +89,8 @@ ExternalProject_Add(
              -DLLVM_DIR=${LLVM_BINARY_DIR}
              -DCLANG_INCLUDE_DIRS=${CLANG_INCLUDE_DIRS}
              ${_clad_extra_cmake_args}
-  BUILD_COMMAND ${CMAKE_COMMAND} --build . --config ${_clad_build_type}
-  INSTALL_COMMAND ${CMAKE_COMMAND} --build . --config ${_clad_build_type} --target install
+  BUILD_COMMAND ${CMAKE_COMMAND} --build . ${EXTRA_BUILD_ARGS}
+  INSTALL_COMMAND ${CMAKE_COMMAND} --build . ${EXTRA_BUILD_ARGS} --target install
   BUILD_BYPRODUCTS ${CLAD_BYPRODUCTS}
   ${_clad_cmake_logging_settings}
   # We need the target clangBasic to be built before building clad. However, we
```

Comment (Mouchemhou) [reply interpreter/cling/tools/plugins/clad/CMakeLists.txt]: what about these commands:

  BUILD_COMMAND ${CMAKE_COMMAND} --build . --config ${_clad_build_type}
  INSTALL_COMMAND ${CMAKE_COMMAND} --build . --config ${_clad_build_type} --target install

Comment (bellenot) [reply interpreter/cling/tools/plugins/clad/CMakeLists.txt]: did you read my answer? Just take the diff

**Review by guitargeek — APPROVED** [review]
Thanks! Looks good to me now, thanks for figuring out that the `_clad_build_type` variable is actually not empty.

I checked on my Linux machine and it works correctly now: the `EXTRA_BUILD_ARGS` are not set because `_clad_build_type STREQUAL "."`.

**Pull request merged**